domains/SignUp.js

```diff
@@ -41,83 +48,168 @@ const validationSchema = yup.object().shape({
-              formikKey="organization"
-              placeholder="Puente"
-            />
-            {formikProps.isSubmitting ? (
-              <ActivityIndicator />
-            ) : (
-              <Button title="Submit" onPress={formikProps.handleSubmit} />
-            )}
-          </>
-        )}
-      </Formik>
-    </SafeAreaView>
+            }
+            setTimeout(() => {
+              actions.setSubmitting(false);
+            }, 1000);
+          }}
+          validationSchema={validationSchema}
+        >
+          {(formikProps) => (
+            <>
+              <FormInput
+                label="First Name"
+                formikProps={formikProps}
+                formikKey="firstname"
+                placeholder="John"
+                autoFocus
+              />
+              <FormInput
+                label="Last Name"
+                formikProps={formikProps}
+                formikKey="lastname"
+                placeholder="Doe"
+              />
+              <FormInput
+                label="Email"
+                formikProps={formikProps}
+                formikKey="email"
+                placeholder="johndoe@example.com"
+              />
+              <FormInput
+                label="Phone Number"
+                formikProps={formikProps}
+                formikKey="phonenumber"
+                placeholder="123-456-7890"
+              />
+              <FormInput
+                label="Password"
+                formikProps={formikProps}
+                formikKey="password"
+                placeholder="Password Here"
+                secureTextEntry
+              />
+              <FormInput
+                label="Organization"
+                formikProps={formikProps}
+                formikKey="organization"
+                placeholder="Puente"
+              />
+              <Button mode="text" theme={theme} dark color="#3E81FD" style={styles.serviceButton} onPress={showModal}>View the terms and service</Button>
+              <View style={styles.container}>
+                <Text style={styles.serviceText}>
+                  By checking this box, I ackowledge that I have read and
+                  understood the Terms and Service agreement.
+                </Text>
+                <View style={styles.checkbox}>
+                  <Checkbox
+                    disabled={false}
+                    theme={theme}
+                    status={checked ? 'checked' : 'unchecked'}
+                    onPress={() => {
+                      setChecked(!checked);
+                    }}
+                  />
+                </View>
+              </View>
+              {formikProps.isSubmitting ? (
+                <ActivityIndicator />
+              ) : (
+                <Button mode="contained" theme={theme} style={styles.submitButton} onPress={formikProps.handleSubmit}>Submit</Button>
+              )}
+              <Portal theme={theme}>
+                <Modal
```

Comment:
Hey! so we're moving in the direction of domain-driven file structure

Documentation
https://tech.offgrid-electric.com/domain-directory-structure-for-react-apps-why-its-worth-trying-b3855ee77a1e

Documentation
https://css-tricks.com/domain-driven-design-with-react/

this will allow us to have reusable components in the right places. For this modal, I'd move it into the components folder so that we can use it across the application!